contracts/GameJam.sol

```diff
@@ -23,4 +54,26 @@ contract GameJam {
 
     emit CompetitorAdded(competitor);
   }
+  // finish function is a payable used to declare the winner via their address
+  // the function should only be called able when the GameJam is in Progress
+  // otherwise, it would be possible to finish again and again
+  function finish(address winner) public payable
+    onlyAtStage(Stages.JamInProgress)
+    transitionNext {
+
+
+      emit WinnerDeclared(winner);
```

Comment:
Is there other things we want to happen when finish is called ?

Reply:
For the moment I think not, can you think of anything else that should happen?

Reply:
A quick validation that winner is a registered competitor would be good, can we get this from the existing mapping ? 🤔

Reply:
Yeah we can:

```solidity
require(competitors[winner] != address(0x0), "competitor is not registered");
```

Reply:
💎

Reply:
Feel free to resolve this and merge 🏇

Reply:
Example here:

https://github.com/PaddyMc/game-tracker/blob/master/contracts/GameTracker.sol#L40

Reply:
Thank you for the example!
Going to merge and include this in the next PR 👍